package/system/mtd/src/linksys_bootcount.c

```diff
@@ -108,24 +106,37 @@ int mtd_resetbc(const char *mtd)
 	}
 
 	num_bc = mtd_info.size / bc_offset_increment;
+	curr = malloc(bc_offset_increment);
 
-	for (i = 0; i < num_bc; i++) {
-		pread(fd, curr, sizeof(*curr), i * bc_offset_increment);
+	if(curr == NULL) {
+		DLOG_ERR("Failed to allocate %u bytes from memory.", bc_offset_increment);
```

Comment:
Might be just me, but I prefer to keep numerical prints consistent between hex and dec. The rest of the function prints in hex without `0x` prefix, so it might be less confusing if this is in the same format.

Adding `0x` prefix or `h` suffix to the rest of the hex print might be an alternative for readability.